device/coordinate_manager.cpp

```diff
@@ -405,6 +405,26 @@ void CoordinateManager::assert_create_coordinate_manager(
     }
 }
 
+void CoordinateManager::shuffle_tensix_harvesting_mask(const std::vector<uint32_t>& harvesting_locations) {
+    std::vector<uint32_t> sorted_harvesting_locations = harvesting_locations;
```

Comment:
How about 
```
std::vector<std::pair<uint32_t, uint32_t>> sorted_harvesting_locations;
for (int i = 0; i < harvesting_locations.size(); i ++) {
  sorted_harvesting_locations.push_back({harvesting_locations[i], i});
}
// Check if lambda is required.
std::sort(sorted_harvesting_locations.begin(), sorted_harvesting_locations.end(), [](auto &left, auto &right) {
    return left.first < right.first;
});
```
Then below you can write just this instead of having to search again:
```
uint32_t sorted_position = sorted_harvesting_location[pos].second;
```

Reply:
I just saw get_harvested_noc_rows, it is a similar function, but the harvesting returned is in physical coordinates, meaning you don't have to sort here. Whereas your returned harvesting is in logical coordinates, right? Consider what would be better here

Reply:
I think the code you suggested is not doing the same things as the original one, but maybe you referenced that through the other two comments. As you mentioned, new harvesting mask should be in logical coordinates lets say. But what is missing here is maybe being able to retrieve the original harvesting mask, I will add that. I think it is much easier to compute everything coordinate related if we shuffle the mask this way in the beginning